.github/workflows/run-integration-tests.yaml

```diff
@@ -0,0 +1,131 @@
+# ```
+
+on:
+  workflow_call:
+    inputs:
+
+      ##
+      ## Configurations:
+      ##
+
+      test_url:
+        required: true
+        type: string
+        description: "URL of terraui instance to test against, required"
+      environment:
+        required: false
+        type: string
+        default: dev
+        description: "environment config to use, default: dev"
+      tests_to_run:
+        required: false
+        type: string
+        description: "space delimited list of js test files to run, default: run all"
+      target_branch:
+        required: false
+        type: string
+        description: "The branch to run integration tests against"
+
+    # technically, the emails are not secret, but they're stored w/ the private key so
+    # we'll just pull them to avoid potential sync/typo/refresh issues.
+    secrets:
+      LYLE_CLIENT_EMAIL: 
+        required: true
+      LYLE_PRIVATE_KEY: 
+        required: true
+      FIRECLOUD_ACCOUNT_CLIENT_EMAIL:
+        required: true
+      FIRECLOUD_ACCOUNT_PRIVATE_KEY:
+        required: true
+
+env:
+  ENVIRONMENT: ${{ inputs.environment }}
+  TEST_URL: ${{ inputs.test_url }}
+
+jobs:
+  run-integration-tests:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: 'read'
+    steps:
+
+      ##
+      ## Handle Checkout:
+      ##
+
+      - name: "Checkout terra-ui code"
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.target-branch }}
+          repository: databiosphere/terra-ui
+          path: terraui
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: "Run Yarn Install"
+        shell: bash
+        run: cd terraui && yarn install
+
+      # Current (Jul 2023) test implementation requires SA Keys to be made available (lyle-utilts.js, terra-sa-utils.js)
+      - name: "assemble Service Account Keys"
+        run: |
+          echo \
+          LYLE_SA_KEY=$(jq --null-input \
+          --arg client_email "${{ secrets.LYLE_CLIENT_EMAIL }}" \
+          --arg key"${{ secrets.LYLE_PRIVATE_KEY }}" \
+          '{"client_email": $client_email, "key": $key}') >> $GITHUB_ENV
+          
+          echo \
+          TERRA_SA_KEY=$(jq --null-input \
+          --arg client_email "${{ secrets.FIRECLOUD_ACCOUNT_CLIENT_EMAIL }}" \
+          --arg key "${{ secrets.FIRECLOUD_ACCOUNT_PRIVATE_KEY }}" \
+          '{"client_email": $client_email, "key": $key}') >> $GITHUB_ENV
+
+      - name: "Run Integration Tests"
+        shell: bash
+        run: |
+          cd terraui/integration-tests &&
+          yarn test ${{ inputs.tests_to_run }}
+
+      - name: 'Upload Test Artifacts'
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: terra-ui-logs-${{ github.run_id }}
+          path: |
+            /tmp/test-results
+            ./terraui/integration-tests/test-results/screenshots
```

Comment:
Just checking, is this capturing the logs as well as the screenshots?

Reply:
I see a bunch of logs in the /tmp folder, but I think it looks correct

example artifact if you want to look yourself
https://github.com/broadinstitute/monkey/suites/14146347486/artifacts/792371713

log screenshot example:
<img width="1263" alt="Screenshot 2023-07-10 at 11 26 08 AM" src="https://github.com/DataBiosphere/terra-ui/assets/85525681/1f28f208-8800-4486-9efe-6dfa307c30d9">

Reply:
Nice. Yeah, that's the type of log I was looking for.